src/Forms/ValidatedInput/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { flow } from 'lodash/fp';
+import styles from './index.scss';
+import Input from '../Input';
+import Validations from '../Validations';
+import ValidationsAndLabel from '../ValidationsAndLabel';
+import withForm from '../withForm';
+import withStyles from '../../../util/withStyles';
+
+function ValidatedInput({
+  field,
+  children,
+  label,
+  name,
+  id,
+  type = 'text',
+  form: { onFieldChange, data },
+  ...otherProps
+}) {
+  const props = { field, label };
+  const Parent = label === undefined ? Validations : ValidationsAndLabel;
+
+  return (
+    <Parent {...props}>
+      <div className="input-wrapper">
```

Comment:
When I prematurely started working on the more advanced inputs I thought that probably it might be a good idea to either wrap all inputs (i.e. put a wrapper inside `Input` or have a special `WrappedInput`.
Right now you'd only be able to create an input with icons by duplicating this code here in the component you want to write. Would be could to always be able to pass a bunch of children with their own absolute positioning into a standard `Input` component instead.